### PR TITLE
Use golangci-lint

### LINKED ISSUE #35: Use golangci-lint for static scanning
In other repositories, we are scanning by running a set of tools and have decided in the community meeting from [2021/11/22](https://github.com/shipwright-io/community/issues/42#issuecomment-975581815) that we want to move to [golangci-lint](https://golangci-lint.run/). Once that is done, we should run the same here as well.

Sibling issues:

- https://github.com/shipwright-io/build/issues/948
- https://github.com/shipwright-io/cli/issues/77

### PR DESCRIPTION
**Pull request by vancanhuit**
# Changes

Closes #35 

# Submitter Checklist

- [ ] Includes tests if functionality changed/was added
- [ ] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/master/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,26 +23,12 @@ jobs:
...
-        - name: Verify fmt
-          run: make verify-fmt
-        - name: Verify generated code
-          run: make verify-generate
-        - name: Verify generated manifests
-          run: make verify-manifests
-        - name: Verify OLM bundle
-          run: make verify-bundle
+        - name: Run golangci-lint
+          uses: golangci-lint/golangci-lint-action@v2
```
Comment (SaschaSchwarze0) [inline .github/workflows/ci.yml:31]:
```suggestion
          uses: golangci/golangci-lint-action@v2
          with:
            args: --timeout=10m```

**Review by SaschaSchwarze0 — CHANGES_REQUESTED** [review]
Correcting the repository name and adding the timeout like in the CLI.

**Review by SaschaSchwarze0 — CHANGES_REQUESTED** [review]
@vancanhuit golintci-lint ran. A few findings to be fixed.

**Comment by vancanhuit** [comment]
Do we need to temporarily disable some rules to make it pass?

**Comment by SaschaSchwarze0** [comment]
> Do we need to temporarily disable some rules to make it pass?

I'd say no. None of the findings sounds complicated to be addressed. Let us know if you find one where you need help.

**Comment by vancanhuit** [comment]
Is there an environmental issue in the `e2e` run?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,26 +27,14 @@ jobs:
...
         steps:
-        - name: Install Go
-          uses: actions/setup-go@v2
-          with:
-            go-version: ${{ matrix.go-version }}
-        - name: Check out code
+        - name: Checkout code
           uses: actions/checkout@v2
-        - name: Verify fmt
-          run: make verify-fmt
```
Comment (adambkaplan) [inline .github/workflows/ci.yml:39]: Does golangci-lint verify go formatting?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -23,26 +27,14 @@ jobs:
...
-        - name: Verify generated code
-          run: make verify-generate
-        - name: Verify generated manifests
-          run: make verify-manifests
-        - name: Verify OLM bundle
-          run: make verify-bundle
```
Comment (adambkaplan) [inline .github/workflows/ci.yml:45]: These checks need to stay - we have client/deepcopy functions that are auto-generated, and the YAML manifests for the bundle need to be kept in sync.

**Review by adambkaplan — CHANGES_REQUESTED** [review]
@vancanhuit I also noticed that you used a merge commit to bring in changes from main. If possible, can you try rebasing instead?

**Comment by vancanhuit** [comment]
> @vancanhuit I also noticed that you used a merge commit to bring in changes from main. If possible, can you try rebasing instead?

Yes, I will try.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (vancanhuit) [reply .github/workflows/ci.yml]: I've pushed a new commit that rebased on latest main and restore those checks.

**Comment by adambkaplan** [comment]
@vancanhuit this PR is failing tests for the same reason I outlined in https://github.com/shipwright-io/operator/pull/41#issuecomment-1005841805. Unfortunately you are stuck until #52 merges (and you will need to rebase afterwards :crying_cat_face:).

**Comment by adambkaplan** [comment]
@vancanhuit #52 has now merged. Please rebase this pull request on top of the current `main` branch, then I think the e2e tests will be fixed.

**Comment by vancanhuit** [comment]
@adambkaplan I rebased this PR. Please check if it is ok.

**Review by adambkaplan — APPROVED** [review]
/approve

Looks good to merge. Ideally we have a make target that lets contributors run this locally - we can follow up in a separate pull request.

**Comment by adambkaplan** [comment]
Verify failed:

```
Verify check failed - run `make generate` and commit the resulting changes to git.
+ echo 'Verify check failed - run `make generate` and commit the resulting changes to git.'
+ exit 1
make: *** [Makefile:124: verify-generate] Error 1
```

**Comment by vancanhuit** [comment]
CI now is green @adambkaplan :+1:

**Review by adambkaplan — APPROVED** [review]
/approve

/lgtm

**Pull request merged**